x-pack/plugins/upgrade_assistant/public/application/components/overview/logs_step/logs_step.tsx

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+
+import { EuiText, EuiSpacer, EuiButton, EuiCallOut, EuiLoadingContent } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedDate, FormattedTime, FormattedMessage } from '@kbn/i18n/react';
+import type { EuiStepProps } from '@elastic/eui/src/components/steps/step';
+
+import { DEPRECATION_LOGS_INDEX } from '../../../../../common/constants';
+import { WithPrivileges } from '../../../../shared_imports';
+import { useAppContext } from '../../../app_context';
+import { loadLogsCheckpoint } from '../../../lib/logs_checkpoint';
+import type { OverviewStepProps } from '../../types';
+import { useDeprecationLogging } from '../../es_deprecation_logs';
+
+const i18nTexts = {
+  logsStepTitle: i18n.translate('xpack.upgradeAssistant.overview.logsStep.title', {
+    defaultMessage: 'Address deprecation logs',
+  }),
+  logsStepDescription: i18n.translate('xpack.upgradeAssistant.overview.logsStep.description', {
+    defaultMessage:
+      'Review the Elasticsearch deprecation logs to make sure you are not using deprecated APIs.',
+  }),
+  viewLogsButtonLabel: i18n.translate(
+    'xpack.upgradeAssistant.overview.logsStep.viewLogsButtonLabel',
+    {
+      defaultMessage: 'View logs',
+    }
+  ),
+  enableLogsButtonLabel: i18n.translate(
+    'xpack.upgradeAssistant.overview.logsStep.enableLogsButtonLabel',
+    {
+      defaultMessage: 'Enable logs',
```

Comment:
"Enabled logging" sounds a bit more natural to me.
```suggestion
      defaultMessage: 'Enable logging',
```